### ISSUE TITLE
Threads stuck in TIMED_WAIT state when calling MultiProducerSequencer::next

### ISSUE THREAD

**Issue by cmuramoto** [LMAX-Exchange/disruptor#172]
Hi, I'm reposting this as an issue, which originated from a comment associated to #164. 

I don't believe that there's a bug in the code, but I hope this issue helps in finding specific patterns/configurations to reproduce the aforementioned behavior in the title.

Essentially 1 out of 7 servers in my production environment, started to have troubles with **MultiProducerSequencer** (disruptor-version 3.3.4). The only difference between the servers is the JVM version, which is @ 1.7.0_60 on the OK ones and 1.7.0_76, everything else is the same (16-core Xeon E7440, 32GB RAM, RHEL 5.5).

I'm using RingBuffer with WorkHandler, essentially in a Many (producer) to One (consumer). The number of consumers is configured to be the same as the number of CPUs.

```java
public EventProcessor(){
//this implements EventFactory, ThreadFactory, ExceptionHandler, WorkHandler
disruptor = new Disruptor<>(this, cap, this, ProducerType.MULTI, new SleepingWaitStrategy());
//
disruptor.setDefaultExceptionHandler(this);
disruptor.handleEventsWithWorkerPool(createWorkers());
}

//creates as many workers as available CPUs
private WorkHandler<EventHolder>[] createWorkers() {
		final WorkHandler[] rv = new WorkHandler[Runtime.getRuntime().availableProcessors()];
		for (int i = 0; i < rv.length; i++) {
			rv[i] = this;
		}

		return rv;
}

//prevent Worker Threads from dying
public void handleEventException(final Throwable ex, final long sequence, final EventHolder event) {
		log.warn(ex.getMessage(), ex);
}
```

Stack dump shows several ajp Threads (which are used by the load balancer) in a similar state:

```
ajp-172.16.100.5-8009-1 tid=355 [WAITING] [DAEMON] (Extracted from heap dump on yjp profiler. JStack shows similar output. YJP does not translate Thread States properly 673=>TIMED_WAITING) 
java.util.concurrent.locks.LockSupport.parkNanos(long) LockSupport.java:349
com.lmax.disruptor.MultiProducerSequencer.next(int) MultiProducerSequencer.java:136
com.lmax.disruptor.MultiProducerSequencer.next() MultiProducerSequencer.java:105
com.lmax.disruptor.RingBuffer.publishEvent(EventTranslatorTwoArg, Object, Object) RingBuffer.java:509
---<omitted>---.java:130
---<omitted>---.java:110
---<omitted>---.java:466
---<omitted>---:355
---<omitted>---:173
org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(ServletRequest, ServletResponse) FilterChainProxy.java:342
---<omitted>---:62
org.springframework.security.web.FilterChainProxy.doFilter(ServletRequest, ServletResponse, FilterChain) FilterChainProxy.java:160
---<omitted>---:37
org.jboss.web.tomcat.security.SecurityAssociationValve.invoke(Request, Response) SecurityAssociationValve.java:182
org.terracotta.modules.tomcat.tomcat_5_5.SessionValve55.tcInvoke(Request, Response) SessionValve55.java:100
org.terracotta.modules.tomcat.tomcat_5_5.SessionValve55.invoke(Request, Response) SessionValve55.java:85
org.terracotta.session.ModernTomcatSessionValve.invoke(Request, Response) ModernTomcatSessionValve.java:66
java.lang.Thread.run() Thread.java:745
```

I collected 2 heap dumps, with a 20min gap between them, and every sequence (cursor,gatingSequenceCache, sequences of gatingSequences) in the **MultiProducerSequencer** were exactly the same. RingBuffer's entries were all empty (My events hold 2 states, which are set by EventTranslatorTwoArg and set to null after processing).

I don't know if there's some kind of JVM-JIT related issue, but after setting **-XX:-UseLoopPredicate** on the problematic server, the issue was gone (well, it didn't present itself so far in about a week).

**Comment by mikeb01** [comment]
There was a number of concurrency issues with earlier builds of the 1.7 jdk.  I would recommend updating to the latest build revision of 1.7 or even 1.8 if you can.

**Comment by lonelysunshine** [comment]
All of the 22 servers in our production environment have the same problem.
disruptor version: 3.3.4
CentOS release 6.5 (Final)
JDK 1.7.0_80

**Comment by mikeb01** [comment]
Do you have a stack trace of the WorkHandler threads.

**Comment by mikeb01** [comment]
BTW, from a tuning perspective use the number of available processors is probably not the best option.  This will mean that system will have few CPUs than the number of threads used for processing.  I.e. application threads = publisher threads + worker threads.  If worker threads = # of CPUs then the publisher threads will be contending with the worker threads for CPU resource.

**Comment by lonelysunshine** [comment]
upload my stack trace and 
/--------
[jstack.log.zip](https://github.com/LMAX-Exchange/disruptor/files/517563/jstack.log.zip)

**Comment by mikeb01** [comment]
This is fixed in 3.3.6.
